CMakeLists.txt

```diff
@@ -51,6 +52,10 @@ set(OPENANN_BRIEF_DESCRIPTION "An open source library for artificial neural netw
 current_time(OPENANN_TIME)
 set(OPENANN_OPTIMIZATION_FLAGS)
 set(OPENANN_PYTHON_COMPILE_ARGS)
+if(CPP_17)
+	compiler_check_add_flag("-std=c++17")
+	compiler_check_add_flag("-Dregister=") # a work-around to get rid of the register keyword
```

Comment:
I think we can delete register from the whole codebase. It is not really required I guess.

Reply:
Unfortunately `lib/CMA-ESpp/cma-es/utils.h` uses `register` in the header, so the work-around seems to be needed.